src/dataprocessor/property/gt_intproperty.h

```
@@ -68,62 +68,6 @@ class GT_DATAMODEL_EXPORT GtIntProperty : public GtProperty<int>
                   int boundary,
                   int value = 0);
 
-    /**
-     * @brief GtIntProperty
-     * @param ident
-     * @param name
-     * @param brief
-     * @param unitCategory
-     * @param value
-     */
-    [[deprecated("Int Properties do not support units properly and "
-                 "will be removed in the future")]]
-    GtIntProperty(const QString& ident,
```

Note removing this can lead to hard-to-find side-effects:

Lets assume the following user code

    m_intProp("intProp", "Test Int", QString(), GtUnit::Category::Angle)
    
After removal of this constructor, the other constructor 

        GtIntProperty(const QString& ident,
                  const QString& name,
                  const QString& brief,
                  int value);

will be silently used, resulting in a value != 0.

@mariusalexander Do you have an idea, what we can do?

---

One idea is to convert the Category namespace in a named enum. Then it should not be possible to use implicit conversion to int, right?

---

I like the idea of using a class enum.

An alternative solution would be to delete the "false" constructor explicitly:

```cpp
GtIntProperty(const QString& ident,
              const QString& name,
              const QString& brief,
              GtUnit::Category cat) = delete;
 ```
 
 This should work as expected

---

I had a quick test. Replacing the enum with a named enum requires many code (trivial) code changes in core and this probably also in other modules. 

So this is unfortunately a no go for a 2.1 release .

I'll try to explicitly disable the old deprecated constructor.